javascript/ql/src/semmle/javascript/security/dataflow/UnsafeJQueryPlugin.qll

```diff
@@ -36,7 +36,7 @@ module UnsafeJQueryPlugin {
       // prefixing prevents forced html/css confusion:
 
       // prefixing through concatenation:
-      StringConcatenation::getFirstOperand(succ) != pred
+      StringConcatenation::getOperand(succ, [1..StringConcatenation::getNumOperand(succ) - 1]) = pred
```

Comment:
👍  quite happy to see this use of `!=` go away, regardless of performance.

Maybe it would be better to use the `taintStep` predicate exposed by the library, though?
```
StringConcatenation::taintStep(src, dst, _, any(int i | i >= 1))
```

Reply:
Yeah, that seems cleaner. Let me do a quick sanity check of the resulting performance..

Reply:
https://git.semmle.com/esben/dist-compare-reports/tree/js/unsafe-jquery_1580812266407 the reformulation does not time out, so it is probably good to go.